pkg/commands/serve/command.go

```diff
@@ -22,7 +24,12 @@ func Command() *cobra.Command {
 			return signals.Do(context.Background(), func(ctx context.Context) error {
 				// track errors
 				var httpErr, grpcErr error
-				func(ctx context.Context) {
+				err := func(ctx context.Context) error {
+					// create a rest config
+					config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
```

Comment:
do you always want to build the config from a kubeconfig? Also when you run it in cluster directly?

Reply:
when run in cluster the `--kubeconfig` flag should not be set (it will resolve to in-cluster config).